src/osd/PG.cc

```diff
@@ -1691,7 +1691,23 @@ std::optional<requested_scrub_t> PG::validate_scrub_mode() const
   return upd_flags;
 }
 
-void PG::on_scrub_schedule_input_change()
+void PG::reschedule_scrub()
+{
+  dout(20) << fmt::format(
+		  "{} for a {}", __func__,
+		  (is_primary() ? "Primary" : "non-primary"))
+	   << dendl;
+
+  // we are assuming no change in primary status
+  if (is_primary()) {
+    ceph_assert(m_scrubber);
+    m_scrubber->update_scrub_job(m_planned_scrub);
+  } else {
+    dout(20) << __func__ << ": inactive or non-primary" << dendl;
+  }
+}
+
+oid PG::on_scrub_schedule_input_change()
```

Comment:
oid?

Reply:
Strange.. this thing ran dozens of tests. I probably ruined something when merging back.